mrp_multi_level/wizards/mrp_multi_level.py

```diff
@@ -0,0 +1,740 @@
+                    product_mrp_area_id, bomline, qty, mrp_date_demand_2, bom, name
+                )
+                mrpmove_id2 = self.env["mrp.move"].create(move_data)
+                if hasattr(action, "mrp_move_down_ids"):
+                    action.mrp_move_down_ids = [(4, mrpmove_id2.id)]
+        return True
+
+    @api.model
+    def create_action(self, product_mrp_area_id, mrp_date, mrp_qty, name, values=None):
+        if not values:
+            values = {}
+        if not isinstance(mrp_date, date):
+            mrp_date = fields.Date.from_string(mrp_date)
+        action_date, date_supply = self._get_action_and_supply_dates(
+            product_mrp_area_id, mrp_date
+        )
+        return self.create_planned_order(
+            product_mrp_area_id, mrp_qty, name, date_supply, action_date, values=values
+        )
+
+    @api.model
+    def create_planned_order(
+        self,
+        product_mrp_area_id,
+        mrp_qty,
+        name,
+        mrp_date_supply,
+        mrp_action_date,
+        values=None,
+    ):
+        self = self.with_context(auditlog_disabled=True)
+        if self._exclude_from_mrp(
+            product_mrp_area_id.product_id, product_mrp_area_id.mrp_area_id
+        ):
+            values["qty_ordered"] = 0.0
+            return values
+
+        qty_ordered = values.get("qty_ordered", 0.0) if values else 0.0
+        qty_to_order = mrp_qty
+        while qty_ordered < mrp_qty:
+            qty = product_mrp_area_id._adjust_qty_to_order(qty_to_order)
+            qty_to_order -= qty
+            order_data = self._prepare_planned_order_data(
+                product_mrp_area_id, qty, mrp_date_supply, mrp_action_date, name
+            )
+            planned_order = self.env["mrp.planned.order"].create(order_data)
+            qty_ordered = qty_ordered + qty
+
+            if product_mrp_area_id._to_be_exploded():
+                self.explode_action(
+                    product_mrp_area_id, mrp_action_date, name, qty, planned_order
+                )
+
+        values["qty_ordered"] = qty_ordered
+        log_msg = "[{}] {}: qty_ordered = {}".format(
+            product_mrp_area_id.mrp_area_id.name,
+            product_mrp_area_id.product_id.default_code
+            or product_mrp_area_id.product_id.name,
+            qty_ordered,
+        )
+        logger.debug(log_msg)
+        return values
+
+    @api.model
+    def _mrp_cleanup(self, mrp_areas):
+        logger.info("Start MRP Cleanup")
+        domain = []
+        if mrp_areas:
+            domain += [("mrp_area_id", "in", mrp_areas.ids)]
+        self.env["mrp.move"].search(domain).unlink()
+        self.env["mrp.inventory"].search(domain).unlink()
+        domain += [("fixed", "=", False)]
+        self.env["mrp.planned.order"].search(domain).unlink()
+        logger.info("End MRP Cleanup")
+        return True
+
+    @api.model
+    def _low_level_code_calculation(self):
+        logger.info("Start low level code calculation")
```

Comment:
@DavidBForgeFlow  Thanks to port this module. 
I think it could be interesting to extract this function to a standalone module because it has few dependencies (only llc field) and this concept could be used in many other uses.

i.e: if you want to recompute cost price of all your products, it's required to begin by those with a higher level code first

Also in a separate module it could "spread the word" to more people of this fundamental concept. (module name could be mrp_low_level_code or any other)

Ok that's another task than migration and may could be done in another module but I need to know if others are on the same line than me.

What do you think ? cc @LoisRForgeFlow @hparfr @dreispt @florian-dacosta @pedrobaeza